### ISSUE TITLE
ec2_tag module doesn't support new aws region ca-central-1

### ISSUE THREAD

**Issue by conlon** [ansible/ansible#20195]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ec2_tag module

##### ANSIBLE VERSION
tested with
```
ansible 2.0.2.0
  config file = /Users/mconlon/git/hyperwave_relay_canada/ansible.cfg
  configured module search path = custom_modules
```
and
```
ansible 2.3.0 (devel 02b548dba2) last updated 2017/01/12 12:21:15 (GMT -500)
  config file = /Users/mconlon/git/hyperwave_relay_canada/ansible.cfg
  configured module search path = [u'custom_modules']
```

(I wiped away my boto installation and reinstalled to get the latest version...also reinstalled boto from scratch when testing devel branch of ansible inside virtualenv.)

##### CONFIGURATION
my ansible.cfg:
```
[defaults]
roles_path = roles
lookup_plugins = lookup_plugins
filter_plugins = filter_plugins
hash_behaviour = merge
library = custom_modules
forks = 50

[ssh_connection]
scp_if_ssh = True
pipelining = True
# an attempt to fix ssh unreachable errors when machines are actually reachable
retries = 5
```

##### OS / ENVIRONMENT
Running from Mac OS X 10.12.2, managing Ubuntu 14.04

##### SUMMARY
The ec2_tag module isn't compatible with ca-central-1 region of AWS

##### STEPS TO REPRODUCE
example tasks which fail (region has been set in the inventory to "ca-central-1"):
```yaml
- name: set role tag
  ec2_tag:
    region: "{{ region }}"
    resource: "{{ ec2_id }}"
    tags:
      role: "{{ ec2_role }}"
```
and
```yaml
- name: check aws Name tag
  ec2_tag:
    region: "{{ region }}"
    resource: "{{ ansible_ec2_instance_id }}"
    state: list
  register: ec2_tags
```

##### EXPECTED RESULTS
The tag should be set on an instance in Canada, or I should be able to get a list of tags on an instance in Canada.

##### ACTUAL RESULTS
The module fails because it doesn't recognize ca-central-1

output:
```
TASK [bastion : set role tag] ***************************************************************************************************************************************************************
task path: /Users/mconlon/git/hyperwave_relay_canada/roles/bastion/tasks/main.yml:39
Using module file /Users/mconlon/ansible_devel/src/ansible/lib/ansible/modules/cloud/amazon/ec2_tag.py
<10.3.1.144> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<10.3.1.144> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o Port=4242 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o 'ProxyCommand=ssh -W %h:%p bastion-01.edge.aworks.ca' -o ControlPath=/Users/mconlon/.ansible/cp/ansible-ssh-%h-%p-%r 10.3.1.144 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
fatal: [10.3.1.144]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            "aws_access_key": null,
            "aws_secret_key": null,
            "ec2_url": null,
            "profile": null,
            "region": "ca-central-1",
            "resource": "i-0e8fb7c78637b5004",
            "security_token": null,
            "state": "present",
            "tags": {
                "role": "bastion"
            },
            "validate_certs": true
        },
        "module_name": "ec2_tag"
    },
    "msg": "Region ca-central-1 does not seem to be available for aws module boto.ec2. If the region definitely exists, you may need to upgrade boto or extend with endpoints_path"
}
```

**Comment by willthames** [comment]
What version of boto are you  using? `python -c 'import boto; print boto.__version__'`

Looks like ca-central was added in 2.44 (latest version is 2.45).

**Comment by conlon** [comment]
@willthames I mentioned above that I wiped away my boto installation and installed the latest version...but yeah just to be 100% clear I'm on 2.45

**Comment by willthames** [comment]
@conlon, sure, but that makes no sense :)

Is it just ec2_tag or does the ec2 module have the same problem?

How about e.g. ec2_vpc_net_facts: `ansible -m ec2_vpc_net_facts -a 'region=ca-central-1' localhost`

There is nothing in `ec2_tag` itself (or indeed in most Ansible AWS modules) that would cause this, and the error appears to be coming from boto - I just can't see why.

Could you run it under `-vvv` and paste the whole task - my only theory is that you're using a different python to the one that boto 2.45 is installed under, and it's picking up an older library.

**Comment by conlon** [comment]
@willthames I agree, and that's why I'm here...I couldn't make sense of it, or find anything particularly weird with the `ec2_tag` module.

The `ec2` module is working just fine, I'm provisioning plenty of instances in ca-central-1.
`ansible -m ec2_vpc_net_facts -a 'region=ca-central-1' localhost` works fine as well.

I'm not as familiar with adhoc mode, so I just tried it with one of my ec2_tag tasks, and that works fine locally:
`ansible -m ec2_tag -a 'region=ca-central-1 resource=i-016e115b681dbd2bb state=list' localhost`

However, I often have this task running from a remote instance, with `resource={{ ansible_ec2_instance_id }}` as retrieved from `ec2_facts` module.

I'm trying to figure out how to verify that the adhoc command above works on a remote host from a dynamic inventory...but at least I've got somewhere to hunt now...
